cranelift/codegen/src/opts/inst_specs_opt.isle

```diff
@@ -0,0 +1,693 @@
+;; from: arithmetic.isle
+;; rewrites for integer and floating-point arithmetic
+;; eg: `iadd`, `isub`, `ineg`, `imul`, `fadd`, `fsub`, `fmul`
+
+;; For commutative instructions, we depend on cprop.isle pushing immediates to
+;; the right, and thus only simplify patterns like `x+0`, not `0+x`.
+
+;; x+0 == x.
+(rule iadd_zero (simplify (iadd ty
```

Comment:
Let's move these back to their original file (as in, put the rule names on the original rules) before we merge